### DISCUSSION TITLE
I have some problem with useSWRInfinite. is this BUG?

### DISCUSSION THREAD

**Discussion by HenryNoh** [vercel/swr#1298] [Q&A]
we are using the useSWRInfinite in our project.
This is an example of our code.

`
const {
        data: initialData,
        error: networkErrorMessage,
        isValidating,
        size,
        setSize,
    } = useSWRInfinite(
        (idx, previousData) => {
            if (!isReady) return
            if (previousData?.data?.error?.code === 1) return null
            else if (!payload) return null
            return `api address~~`
        },
        { initialSize: 2 }
    )
    
    if (Number(initialData) === 1) {
        return {
            data: undefined,
            error: undefined,
        }
    }
    
    return {
        data,
        error,
        count,
        size,
        isValidating,
        setSize,
    }
    `
    
In some cases, we get the initialData : 1.
but we already catch that error to make undefined. but its still make some problem. plz help us and Is it BUG in useSWRInfinite?

**Comment by shuding** [comment]
Please fix the format of your code snippets, it’s really hard to read and understand. Thanks!

**Comment by swandogg** [comment]
🐛 it is i
